tests/framework/BaseYiiTest.php

```diff
@@ -84,4 +86,19 @@ public function testCreateObjectCallable()
             return true;
         }));
     }
+
+    /**
+     * @covers yii\BaseYii::setLogger()
+     */
+    public function testSetLogger()
+    {
+        $logger = new Logger();
+        BaseYii::setLogger($logger);
+
+        $reflection = new \ReflectionClass('yii\BaseYii');
```

Comment:
Now your test depends on the implementation of the `setLogger()` method, which is contradicts the unit test paradigma.
If the internal implemenation of the method changes - the test will fail.